### PR TITLE
HHH-17806 Reduce the number of unnecessary joins with `JOINED` inheritance and parameterized discriminator predicates

### PR DESCRIPTION
**Pull request by mbladel**
https://hibernate.atlassian.net/browse/HHH-17806

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java [hunk 1]
```diff
@@ -7590,41 +7590,45 @@ private void handleTypeComparison(
...
 				}
 			}
-			else {
-				final Set<String> excludedEntityNames = new HashSet<>( entityMappingType.getSubMappingTypes().size() );
-				for ( EntityTypeLiteral literalExpr : literalExpressions ) {
-					excludedEntityNames.add( literalExpr.getEntityTypeDescriptor().getEntityName() );
-				}
-				if ( !excludedEntityNames.contains( entityMappingType.getEntityName() ) ) {
-					registerEntityNameUsage( tableGroup, EntityNameUse.FILTER, entityMappingType.getEntityName() );
+			if ( literalExpressions != null ) {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:7610]: Minor, but this is just the `else` branch to the `if ( literalExpressions == null )` check.  No need for another check.

Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:7610]:
```suggestion
			else {
```

**Review by sebersole — APPROVED** [review]
Looks good to me.  

Did leave a minor comment.

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java (hunk 1 above)
Comment (mbladel) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java]: Oops, must have changed that by mistake - thanks!

**Review by beikov — APPROVED** [review]

**Pull request merged**